### PR TITLE
ESQL: Update guide for building scalar functions

### PR DESCRIPTION
**Pull request by nik9000**
This updates the guide for building your first scalar functions, mostly to include a list of the annotations that you can add to your `process` method.

### CODE REVIEW

**Comment by nik9000** [comment]
I'll add some more examples to the linked generation annotations in separate PRs.

**Comment by nik9000** [comment]
I'm going to update this with references to the superclasses for functions.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java [hunk 1]
```diff
@@ -56,20 +56,51 @@
...
- *         generated evaluator before submitting your PR.
- *     <li>
- *         Once your evaluator is generated you can implement
- *         {@link org.elasticsearch.xpack.esql.evaluator.mapper.EvaluatorMapper#toEvaluator},
- *         having it return the generated evaluator.
+ *         that contain the actual inner implementation of the function. They are usually named
+ *         "process" or "processInts" or "processBar". Modify those to look right and click
+ *         {@code Build->Recompile 'FunctionName.java'} in IntelliJ or run the {@code CsvTests}
+ *         again. This should generate an {@link org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator}
+ *         implementation calling the method annotated with {@link org.elasticsearch.compute.ann.Evaluator}.
```
Comment (ivancea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java:77]: > click Build->Recompile 'FunctionName.java' in IntelliJ or run the  CsvTests again

You actually need to run both (One for IntelliJ to recognize the files, and the other to generate the real files. As IntelliJ and Gradle generate them in different locations:

image

So I would maybe go with something like:
```suggestion
 *         "process" or "processInts" or "processBar". Modify those to look right and run the {@code CsvTests} again. This should generate an {@link org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator}
 *         implementation calling the method annotated with {@link org.elasticsearch.compute.ann.Evaluator}.
 *.        To make it work with IntelliJ, also click {@code Build->Recompile 'FunctionName.java'}.
```

I don't know if we can fix it instead, would be nice. [But I see you tried it already](https://github.com/elastic/elasticsearch/pull/107313/files#diff-53f3a207a751d5038b9c7211709242c1fe2c4668e79d33f6b9ff5b04dcfea446R46)

**Review by ivancea — APPROVED** [review]
Looks good!

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java (hunk 1 above)
Comment (nik9000) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java:77]: Yeah, if you want to fight with the build system and intellij, be my guest. They do not agree.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java [hunk 2]
```diff
@@ -8,7 +8,7 @@
 /**
  * Functions that take a row of data and produce a row of data without holding
  * any state between rows. This includes both the {@link org.elasticsearch.xpack.esql.core.expression.function.scalar.ScalarFunction}
- * subclass to link into the QL infrastucture and the {@link org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator}
+ * subclass to link into the QL infrastructure and the {@link org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator}
```
Comment (bpintea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java:11]: Should this be updated to:
```suggestion
 * subclass to link into the ESQL core infrastructure and the {@link org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator}
```

**Review by bpintea — APPROVED** [review]
Lgtm

**Pull request merged**